corehq/warehouse/models/facts.py

```diff
@@ -80,41 +80,64 @@ def dependencies(cls):
             FORM_STAGING_SLUG,
         ]
 
-# TODO: Write SyncLogFact
+
+class SyncLogFact(BaseFact, CustomSQLETLMixin):
+    '''
+    SyncLog Fact Table
+    Grain: sync_log_id
+    '''
+    slug = SYNCLOG_FACT_SLUG
+
+    sync_log_id = models.CharField(max_length=255)
+    sync_date = models.DateTimeField(null=True)
+
+    # these can be null per SyncLogStagingTable
+    domain = models.CharField(max_length=255, null=True)
+
+    user_dim = models.ForeignKey(UserDim, on_delete=models.PROTECT)
+    # not all synclogs have domains, added in 11/2016
+    domain_dim = models.ForeignKey(DomainDim, on_delete=models.PROTECT, null=True)
```

Comment:
can we not infer this from the user and populate it anyway?

Reply:
I do not think we can for web users using webapps, but all new synclogs should have this anyway so we would not expect that to be an issue